Installation issue: netcdf-c-4.9.2

### Steps to reproduce the issue

```console
$ spack spec -I netcdf-c@4.9.2 %gcc@12.2.0
 -   netcdf-c@4.9.2+blosc~byterange~dap~fsync~hdf4~jna~logging+mpi~nczarr_zip+optimize~parallel-netcdf+pic+shared+szip+zstd build_system=autotools patches:=0161eb8 arch=linux-centos7-haswell %c=gcc@12.2.0
[+]      ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^diffutils@3.10 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]              ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-centos7-haswell %c=gcc@12.2.0
[+]      ^c-blosc@1.21.6+avx2~ipo build_system=cmake build_type=Release generator=make arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]          ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]              ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]                  ^nghttp2@1.65.0 build_system=autotools arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]              ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]          ^lz4@1.10.0+pic build_system=makefile libs:=shared,static arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]          ^snappy@1.2.1~ipo+pic+shared build_system=cmake build_type=Release generator=make arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]      ^compiler-wrapper@1.0 build_system=generic arch=linux-centos7-haswell
[e]      ^gcc@12.2.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' arch=linux-centos7-haswell
[+]      ^gcc-runtime@12.2.0 build_system=generic arch=linux-centos7-haswell
[e]      ^glibc@2.17 build_system=autotools patches:=be65fec,e179c43 arch=linux-centos7-haswell
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-centos7-haswell %c=gcc@12.2.0
[+]      ^hdf5@1.14.6~cxx~fortran+hl~ipo~java~map+mpi+shared~subfiling~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^pkgconf@2.3.0 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]      ^libaec@1.0.6~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]      ^openmpi@5.0.7+atomics~cuda~debug+fortran~gpfs~internal-hwloc~internal-libevent~internal-pmix~ipv6~java~lustre~memchecker~openshmem~romio+rsh~static~two_level_namespace+vt+wrapper-rpath build_system=autotools fabrics:=none patches:=38529b5 romio-filesystem:=none schedulers:=none arch=linux-centos7-haswell %c,cxx,fortran=gcc@12.2.0
[+]          ^autoconf@2.72 build_system=autotools arch=linux-centos7-haswell
[+]              ^m4@1.4.19+sigsegv build_system=autotools patches:=9dc5fbd,bfdffa7 arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]                  ^libsigsegv@2.14 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^automake@1.16.5 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^hwloc@2.11.1~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl+pci~rocm build_system=autotools libs:=shared,static arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]              ^libpciaccess@0.17 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]                  ^util-macros@1.20.1 build_system=autotools arch=linux-centos7-haswell
[+]              ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]                  ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^libevent@2.1.12+openssl build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]              ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]                  ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-centos7-haswell
[+]          ^libtool@2.4.7 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]              ^findutils@4.10.0 build_system=autotools patches:=440b954 arch=linux-centos7-haswell %c=gcc@12.2.0
[+]                  ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]                      ^tar@1.35 build_system=autotools zip=pigz arch=linux-centos7-haswell %c=gcc@12.2.0
[+]                          ^pigz@2.8 build_system=makefile arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^numactl@2.0.18 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^openssh@9.9p1+gssapi build_system=autotools arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]              ^krb5@1.21.3+shared build_system=autotools arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]                  ^bison@3.8.2~color build_system=autotools arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]              ^libedit@3.1-20240808 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]              ^libxcrypt@4.4.38~obsolete_api build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-centos7-haswell %c=gcc@12.2.0
[+]              ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]              ^gdbm@1.23 build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]                  ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-centos7-haswell %c=gcc@12.2.0
[+]          ^pmix@5.0.5~munge~python build_system=autotools arch=linux-centos7-haswell %c=gcc@12.2.0
[+]      ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
[+]      ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static arch=linux-centos7-haswell %c,cxx=gcc@12.2.0
```


### Error message

<details><summary>Error message</summary>

<pre>
==> No binary for netcdf-c-4.9.2-dthcftimehqhov644v5pidgqaotpojzb found: installing from source
==> Installing netcdf-c-4.9.2-dthcftimehqhov644v5pidgqaotpojzb [49/49]
==> Using cached archive: /public1/wshome/ws365/scfa0352/lwh/spack-develop/var/spack/cache/_source-cache/archive/bc/bc104d101278c68b303359b3dc4192f81592ae8640f1aee486921138f7f88cb7.tar.gz
==> Using cached archive: /public1/wshome/ws365/scfa0352/lwh/spack-develop/var/spack/cache/_source-cache/archive/01/0161eb870fdfaf61be9d70132c9447a537320342366362e76b8460c823bf95ca
==> Applied patch https://github.com/Unidata/netcdf-c/commit/f8904d5a1d89420dde0f9d2c0e051ba08d08e086.patch?full_index=1
==> netcdf-c: Executing phase: 'autoreconf'
==> netcdf-c: Executing phase: 'configure'
==> netcdf-c: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_pl/linux-haswell/gmake-4.4.1-d5otmmzp474qf3rusilqnrglaqz64xdz/bin/make' '-j24' 'V=1'

3 errors found in build log:
     699    libtool: link: ar cr .libs/libnetcdf2.a .libs/libnetcdf2_la-dv2i.o
     700    libtool: link: ranlib .libs/libnetcdf2.a
     701    libtool: link: ( cd ".libs" && rm -f "libnetcdf2.la" && ln -s "../libnetcdf2.la" "libnetcdf2.la" )
     702    mv -f .deps/libdispatch_la-utf8proc.Tpo .deps/libdispatch_la-utf8proc.Plo
     703    /bin/sh ../libtool  --tag=CXX   --mode=link g++    -lstdc++ -L/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_
            pl/linux-haswell/hdf5-1.14.6-miimhnns3qaqwbimbezzno2uiew3novb/lib -L/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack
            _path_pl/linux-haswell/zlib-ng-2.2.4-cyqxdhuxn7ywxd5myokymzxfpdzqlimo/lib -L/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__
            /__spack_path_pl/linux-haswell/libaec-1.0.6-xcxrsybxejxuhioww3eciztww3fiih5i/lib64 -L/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spack_path_plac
            eholder__/__spack_path_pl/linux-haswell/bzip2-1.0.8-jtk37k4qzcicyr3j52e3vovx56x3z22p/lib -L/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spack_pat
            h_placeholder__/__spack_path_pl/linux-haswell/zstd-1.5.7-lg7oxdsb7bgvwij2nfxqnf4phipiw4qw/lib -L/public1/wshome/ws365/scfa0352/lwh/spack-develop/opt/spack/__spack_path_placeholder__/__spac
            k_path_placeholder__/__spack_path_pl/linux-haswell/c-blosc-1.21.6-wrucberrvpdevqoxgdfvm3ldvn4wofvq/lib64 -o libdispatch.la  libdispatch_la-dcopy.lo libdispatch_la-dfile.lo libdispatch_la-d
            dim.lo libdispatch_la-datt.lo libdispatch_la-dattinq.lo libdispatch_la-dattput.lo libdispatch_la-dattget.lo libdispatch_la-derror.lo libdispatch_la-dvar.lo libdispatch_la-dvarget.lo libdis
            patch_la-dvarput.lo libdispatch_la-dvarinq.lo libdispatch_la-dinternal.lo libdispatch_la-ddispatch.lo libdispatch_la-dutf8.lo libdispatch_la-nclog.lo libdispatch_la-dstring.lo libdispatch_
            la-ncuri.lo libdispatch_la-nclist.lo libdispatch_la-ncbytes.lo libdispatch_la-nchashmap.lo libdispatch_la-nctime.lo libdispatch_la-nc.lo libdispatch_la-nclistmgr.lo libdispatch_la-dauth.lo
             libdispatch_la-doffsets.lo libdispatch_la-dpathmgr.lo libdispatch_la-dutil.lo libdispatch_la-dreadonly.lo libdispatch_la-dnotnc4.lo libdispatch_la-dnotnc3.lo libdispatch_la-dinfermodel.lo
             libdispatch_la-daux.lo libdispatch_la-dinstance.lo libdispatch_la-dcrc32.lo libdispatch_la-dcrc64.lo libdispatch_la-ncexhash.lo libdispatch_la-ncxcache.lo libdispatch_la-ncjson.lo libdisp
            atch_la-ds3util.lo libdispatch_la-dparallel.lo libdispatch_la-utf8proc.lo libdispatch_la-drc.lo libdispatch_la-dgroup.lo libdispatch_la-dvlen.lo libdispatch_la-dcompound.lo libdispatch_la-
            dtype.lo libdispatch_la-denum.lo libdispatch_la-dopaque.lo libdispatch_la-dfilter.lo     -lhdf5_hl -lhdf5 -lm -lz -ldl -lsz -lbz2 -lzstd -lblosc
     704    libtool:   error: unsupported hardcode properties See the libtool documentation for more information. Fatal configuration error.
  >> 705    make[2]: *** [Makefile:653: libdispatch.la] Error 1
     706    make[2]: Leaving directory '/tmp/scfa0352/spack-stage/spack-stage-netcdf-c-4.9.2-dthcftimehqhov644v5pidgqaotpojzb/spack-src/libdispatch'
  >> 707    make[1]: *** [Makefile:769: all-recursive] Error 1
     708    make[1]: Leaving directory '/tmp/scfa0352/spack-stage/spack-stage-netcdf-c-4.9.2-dthcftimehqhov644v5pidgqaotpojzb/spack-src'
  >> 709    make: *** [Makefile:614: all] Error 2

See build log for details:
  /tmp/scfa0352/spack-stage/spack-stage-netcdf-c-4.9.2-dthcftimehqhov644v5pidgqaotpojzb/spack-build-out.txt

</pre></details>


### Information on your system

* **Spack:** 1.0.0.dev0
* **Builtin repo:** not available
* **Python:** 3.6.8
* **Platform:** linux-centos7-haswell

### Additional information

[spack-build-02-configure-out.txt](https://github.com/user-attachments/files/21071969/spack-build-02-configure-out.txt)
[spack-build-01-autoreconf-out.txt](https://github.com/user-attachments/files/21071968/spack-build-01-autoreconf-out.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/21071965/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/21071966/spack-build-env.txt)
[spack-build-03-build-out.txt](https://github.com/user-attachments/files/21071967/spack-build-03-build-out.txt)
[spack-build-env-mods.txt](https://github.com/user-attachments/files/21071970/spack-build-env-mods.txt)

@skosukhin

### General information

- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform

--- a/repos/spack_repo/builtin/packages/netcdf_c/package.py
+++ b/repos/spack_repo/builtin/packages/netcdf_c/package.py
@@ -381,6 +381,18 @@ class AutotoolsBuilder(AnyBuilder, autotools.AutotoolsBuilder):
     def force_autoreconf(self):
         return any(self.spec.satisfies(s) for s in self.pkg._force_autoreconf_when)
 
+    @property
+    def build_targets(self):
+        # Starting version 4.8.0, the library includes C++ source files. None of those files is
+        # compiled in any configuration that we currently support. However, Automake still chooses
+        # the C++ compiler for linking, which leads to overlinking to the standard C++ library.
+        # To avoid that, we run make with an extra argument, which overrides the linker command.
+        # This way, the C++ compiler is never called, and the linking is done with the default
+        # command that runs the C compiler.
+        if self.spec.satisfies("@4.8.0:"):
+            return ["CXXLINK=${LINK}"]
+        return []
+
     @when("@4.6.3:")
     def autoreconf(self, pkg, spec, prefix):
         if not os.path.exists(self.configure_abs_path):

